### PR TITLE
2.0 migrations

### LINKED ISSUE #4630: Write cart and orders schema migrations for 2.0 changes
## Work
Cart and orders database schemas changed quite a bit in 2.0. Write a migration to convert pre-2.0 data into the new schema.

### PR DESCRIPTION
**Pull request by aldeed**
Resolves #4630
Impact: **minor**  
Type: **feature**

## Changes
Migrate existing data for cart and orders after an upgrade

## Breaking changes
No, quite the opposite hopefully

## Testing
1. Check out `master`
2. Run Reaction and do complete checkout flow for anonymous and logged in. Also add some more items to carts that you do not check out. With at least one of the carts, do the checkout flow but don't click the final place order button. The goal is to have some carts in various stages of data collection, and some orders for different types of users.
    - It can be good to also enable taxes and add some taxes for your shipping postal code, since some of the conversions are related to taxes.
    - I recommend having at least one order paid for with "example" method and at least one paid for with a credit card through Stripe package.
3. Stop Reaction and check out this branch.
4. In the `Migrations` collection in MongoDB, make sure that the control document has `locked === false` and a version at or less than 38.
5. Run Reaction and verify that you see a message on startup that it ran the migrations up to version 40.
6. Smoke test in the app, including running the existing converted orders through all possible statuses.

### CODE REVIEW

**Comment by zenweasel** [comment]
Started on 1.16, created some orders. Migrated to 2.0. If I try to capture an order placed before migration I see this error:

```
   { Error: Tax rate is required (shipping.0.items.0.taxRate) in Orders update [400]
    at getErrorObject (packages/aldeed:collection2/collection2.js:504:28)
    at doValidate (packages/aldeed:collection2/collection2.js:470:13)
```

**Comment by zenweasel** [comment]
I created an account in previous version, then tried to checkout with that account and got a blank block for shipping until I edited and resaved the address. Only happened the first time. Not sure it's necessarily related to this PR.

**Comment by zenweasel** [comment]
Trying to check out with Example payment enabled there a problem with the payment form;

File: imports/plugins/core/versions/server/util/convert20.js [hunk 1]
```diff
@@ -0,0 +1,238 @@
...
+/**
+ * @param {Object} cart The cart document to transform
+ * @returns {Object} The converted cart document
+ */
+export function convertCart(cart) {
+  const convertedCart = { ...cart };
+
+  // Set `cart.billingAddress`
+  if (!cart.billingAddress && Array.isArray(cart.billing) && cart.billing[0]) {
+    convertedCart.billingAddress = cart.billing[0].address;
```
[the remark this answers is not in the document]
Comment (aldeed) [reply imports/plugins/core/versions/server/util/convert20.js]: How do we define default? I believe billing address was previously set on either all items in `billing` array, or just the first, which either way should work for this approach. But I agree it smells bad, so maybe I'll have it use the first item in billing array where `address` is set.

The worst case here is that someone halfway through checkout out will have to re-enter their billing address, so this part is low risk.

**Comment by aldeed** [comment]
@zenweasel 

- `taxRate` error I can fix. Any chance you saved off the order document from the DB before you migrated?
- I think I've seen that address issue on 1.x before. Not sure exactly how it happens.
- That's what the payment form looks like now. It's Example IOU payment type and requires only that you enter a name, to avoid confusing it with a payment package that they should actually use.

**Comment by zenweasel** [comment]
@aldeed I didn't save it but i just created an order with no tax method enabled. Just realized I didn't test marketplace orders. Should I do it again to test it?

**Comment by aldeed** [comment]
That's fine, I figured out the tax rate thing. I don't think we're testing marketplace anymore, although I did write it in a way that should handle marketplace orders.

**Comment by aldeed** [comment]
@zenweasel Pushed fixes for those two issues.

**Comment by zenweasel** [comment]
@aldeed Starting to test now

**Comment by zenweasel** [comment]
Again not sure if this is related to this PR and will try to test but I see this weird thing (notice that tax was not enabled)

http://g.recordit.co/z6VsMIn97w.gif

**Pull request merged**